sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/kafka/AzureEventHubsKafkaOAuth2AutoConfiguration.java

```diff
@@ -0,0 +1,121 @@
+                                               AzureGlobalProperties azureGlobalProperties) {
+        this.kafkaProperties = kafkaProperties;
+        this.tokenCredentialResolver = resolver;
+        this.defaultTokenCredential = defaultTokenCredential;
+        this.azureGlobalProperties = azureGlobalProperties;
+        configureKafkaUserAgent();
+    }
+
+    @Bean
+    DefaultKafkaConsumerFactoryCustomizer azureOAuth2KafkaConsumerFactoryCustomizer() {
+        Map<String, Object> updateConfigs = new HashMap<>();
+        Map<String, Object> consumerProperties = kafkaProperties.buildConsumerProperties();
+        configureOAuth2Properties(updateConfigs, consumerProperties);
+        return factory -> factory.updateConfigs(updateConfigs);
+    }
+
+    @Bean
+    DefaultKafkaProducerFactoryCustomizer azureOAuth2KafkaProducerFactoryCustomizer() {
+        Map<String, Object> updateConfigs = new HashMap<>();
+        Map<String, Object> producerProperties = kafkaProperties.buildProducerProperties();
+        configureOAuth2Properties(updateConfigs, producerProperties);
+        return factory -> factory.updateConfigs(updateConfigs);
+    }
+
+    private void configureOAuth2Properties(Map<String, Object> updateConfigs, Map<String, Object> sourceKafkaProperties) {
+        if (needConfigureSaslOAuth(sourceKafkaProperties)) {
+            AzureKafkaProperties azureKafkaProperties = buildAzureProperties(sourceKafkaProperties,
+                    azureGlobalProperties);
+            updateConfigs.put(AZURE_TOKEN_CREDENTIAL, resolveSpringCloudAzureTokenCredential(azureKafkaProperties));
+            updateConfigs.putAll(KAFKA_OAUTH_CONFIGS);
+            logConfigureOAuthProperties();
+        }
+    }
+
+    private TokenCredential resolveSpringCloudAzureTokenCredential(AzureKafkaProperties azureKafkaConsumerProperties) {
+        TokenCredential tokenCredential = tokenCredentialResolver.resolve(azureKafkaConsumerProperties);
+        return tokenCredential == null ? defaultTokenCredential : tokenCredential;
+    }
+
+    private void configureKafkaUserAgent() {
+        Method dataMethod = ReflectionUtils.findMethod(ApiVersionsRequest.class, "data");
+        if (dataMethod != null) {
+            ApiVersionsRequest apiVersionsRequest = new ApiVersionsRequest.Builder().build();
+            ApiVersionsRequestData apiVersionsRequestData = (ApiVersionsRequestData) ReflectionUtils.invokeMethod(dataMethod, apiVersionsRequest);
+            if (apiVersionsRequestData != null) {
+                apiVersionsRequestData.setClientSoftwareName(apiVersionsRequestData.clientSoftwareName()
+                        + AZURE_SPRING_EVENT_HUBS_KAFKA_OAUTH);
+                apiVersionsRequestData.setClientSoftwareVersion(VERSION);
+            }
+        }
+    }
+
+    @ConditionalOnClass(KafkaMessageChannelBinder.class)
+    @Configuration(proxyBeanMethods = false)
+    @Import(KafkaBinderConfiguration.class)
```

Comment:
Do we need this `import`?

Reply:
yeah, by deafult the BPP won't be loaded into the binder's context's bean factory thus without this import it cannot take effects. while using this import cannot add the bpp into the binder's context either, but it can load the binder configuration into the application context

Reply:
We'd better not do this, the only way to load a `KafkaBinderConfiguration` should be the BinderFactory.